docs/agent/config/v3.mdx

```diff
@@ -462,11 +249,387 @@ These are a list of specifiers for what `Host` headers will be allowed to make r
+  - When using the _domain_ format you are only defining the host.
+    - Scheme will default to `http`.
+    - Port will default to `80`.
+- **Scheme (shorthand)** - `https://`
+  - This only works for `https` and `http`.
+    - For `tcp` and `tls` host and port is required.
+  - When using _scheme_ you are defining the protocol and the port will be inferred on the local host.
+    - For `https` port will be`443`.
+    - For `http` port will be `80`.
+    - Host will be `localhost`.
+- **Port (shorthand)** - `8080`
+  - When using _port_ you are defining the port on the local host that will receive traffic.
+    - Scheme will default to `http`.
+    - Host will default to `localhost`.
+
+###### Example
+
+```yaml {16-17}
+# ...
+endpoints:
+  - name: example
+    url: https://example.ngrok.app
+    metadata: '{ "id": "example-app" }'
+    description: "our example application"
+    traffic_policy:
+      inbound:
+        - actions:
+            - type: custom-response
+              config:
+                status_code: 200
+                content: hello, traffic policy!
+                headers:
+                  content-type: text/plain
+    upstream:
+      url: 8080
+# ...
+```
+
+##### `upstream.protocol`
+
+String. The upstream application layer protocol to be used for HTTP endpoints.
+
+###### Accepted Values:
+
+- `http1` (default)
+- `http2`
+
+###### Example
+
+```yaml {18-18}
+# ...
+endpoints:
+  - name: example
+    url: https://example.ngrok.app
+    metadata: '{ "id": "example-app" }'
+    description: "our example application"
+    traffic_policy:
+      inbound:
+        - actions:
+            - type: custom-response
+              config:
+                status_code: 200
+                content: hello, traffic policy!
+                headers:
+                  content-type: text/plain
+    upstream:
+      url: 8080
+      protocol: http1
+# ...
+```
+
+##### `upstream.proxy_protocol`
+
+String. The version of [PROXY protocol](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) to use with this endpoint, empty if not using.
+
+Example values are `1` or `2`.
+
+###### Example
+
+```yaml {19-19}
+# ...
+endpoints:
+  - name: example
+    url: https://example.ngrok.app
+    metadata: '{ "id": "example-app" }'
+    description: "our example application"
+    traffic_policy:
+      inbound:
+        - actions:
+            - type: custom-response
+              config:
+                status_code: 200
+                content: hello, traffic policy!
+                headers:
+                  content-type: text/plain
+    upstream:
+      url: 8080
+      protocol: http1
+      proxy_protocol: 2
+# ...
+```
+
+## Example Configuration Files
+
+Below are a collection of different agent configurations to serve as examples for your `ngrok.yml` file. The [next section](/docs/agent/config/v3/#endpoint-configurations) explains how to start some or all of your configured endpoints from the CLI.
```

Comment:
I think this was originally my copy-text / _my fault_, this link doesn't seem to go anywhere : |